server/proxy/pf_context.c

```diff
@@ -24,13 +24,59 @@
 #include <winpr/crypto.h>
 #include <winpr/print.h>
 
+#include <freerdp/server/proxy/proxy_log.h>
 #include <freerdp/server/proxy/proxy_server.h>
 
 #include "pf_client.h"
+#include "pf_utils.h"
 #include <freerdp/server/proxy/proxy_context.h>
 
 #include "channels/pf_channel_rdpdr.h"
 
+#define TAG PROXY_TAG("server")
+
+static UINT32 ChannelId_Hash(const void* key)
+{
+	const UINT32* v = (const UINT32*)key;
+	return *v;
+}
+
+static BOOL ChannelId_Compare(const UINT32* v1, const UINT32* v2)
```

Comment:
Why not just use the pointer value? that would eliminate the requirement for custom compare and hash functions.

Reply:
If I'm correct without the custom function it prevents from using `id==0` because the code in hashtable checks for null pointer for the key.

Reply:
true. is `channelID == 0` allowed?